fix(#118): validate the child namespace for RFC 1123

gkarthiks opened the pull request:
Signed-off-by: gkarthiks <github.gkarthiks@gmail.com>

This PR will validate the name of the anchor to make sure the child namespace is in compliance with RFC 1123.

Fixes: #118 

Tested: added unit test, and verified the following error message if I ran this manually: 
> `Subnamespace child.01 is not a valid namespace name: a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')`

/cc @adrianludwin

adrianludwin commented on internal/anchor/validator.go:
```diff
@@ -3,6 +3,7 @@ package anchor
 import (
 	"context"
 	"fmt"
+	"k8s.io/apimachinery/pkg/util/validation"
```
Can you please move this to the next "paragraph" (e.g. line 8 or lower)? The first "paragraph" is for golang standard libraries, the second is for other external dependencies, and the third is for HNC-internal packages.

gkarthiks replied:
done, moved that. I should have noticed that. Apologies that's my bad.

adrianludwin reviewed (commented):
Thanks, this is great! Just two little comments:
1. Rearrange the import order (see other comment for details)
2. Can you please add a bit of information to the commit message about how you tested this? E.g.:

```
Tested: added unit test, and verified the following error message if I ran this manually: 'Subnamespace name is not valid: ...'
```

It would be nice to document somewhere what the message actually looked like, I'm sure you already did run that to check and see but it's nice to write it down.

Please modify the original commit and force-push the branch, we like PRs with only one commit each. Thanks!

gkarthiks commented:
> Thanks, this is great! Just two little comments:
> 
> 1. Rearrange the import order (see other comment for details)
> 2. Can you please add a bit of information to the commit message about how you tested this? E.g.:
> 
> ```
> Tested: added unit test, and verified the following error message if I ran this manually: 'Subnamespace name is not valid: ...'
> ```
> 
> It would be nice to document somewhere what the message actually looked like, I'm sure you already did run that to check and see but it's nice to write it down.
> 
> Please modify the original commit and force-push the branch, we like PRs with only one commit each. Thanks!

Have incorporated the review comments and added the details in the commit message and in the PR desc as well.

adrianludwin commented:
Hey, it looks like something went wrong with your last push? You should only be pushing one commit from your branch, not merging this to your master and then pushing the master. Also it doesn't look like you updated the original commit message? LMK if you're having trouble with this. Thanks.

gkarthiks commented:
> Hey, it looks like something went wrong with your last push? You should only be pushing one commit from your branch, not merging this to your master and then pushing the master. Also it doesn't look like you updated the original commit message? LMK if you're having trouble with this. Thanks.

Apologies on this. Let me create a new PR please.
